packages/grid/src/mouse-handlers/GridLinkMouseHandler.ts

```diff
@@ -0,0 +1,247 @@
+    grid: Grid
+  ): LinkInformation | null {
+    const { column, row, x, y } = gridPoint;
+    const { props, canvasContext: context, renderer, metrics } = grid;
+    const { model } = props;
+
+    if (row == null || column == null) {
+      return null;
+    }
+
+    assertNotNull(context);
+    assertNotNull(metrics);
+    const modelRow = grid.getModelRow(row);
+    const modelColumn = grid.getModelColumn(column);
+    const renderState = grid.getRenderState();
+    const theme = grid.getTheme();
+
+    const text = model.textForCell(modelColumn, modelRow);
+    const {
+      width: textWidth,
+      x: textX,
+      y: textY,
+    } = renderer.getTextRenderMetrics(context, renderState, column, row);
+    const {
+      fontWidths,
+      gridX,
+      gridY,
+      width: gridWidth,
+      height: gridHeight,
+      verticalBarWidth,
+      horizontalBarHeight,
+    } = metrics;
+
+    // Set the font and change it back after
+    const prevFont = context.font;
+    context.font = theme.font;
+
+    const fontWidth =
+      fontWidths.get(context.font) ?? GridRenderer.DEFAULT_FONT_WIDTH;
+    const truncationChar = model.truncationCharForCell(modelColumn, modelRow);
+    const truncatedText = GridRenderer.truncateToWidth(
+      context,
+      text,
+      textWidth,
+      fontWidth,
+      truncationChar
+    );
+
+    // Check if the truncated text contains a link
+    if (linkify.find(truncatedText, 'url').length !== 0) {
+      const tokenizedText = model.tokensForCell(truncatedText);
+      const {
+        actualBoundingBoxAscent,
+        actualBoundingBoxDescent,
+      } = context.measureText(truncatedText);
+      const linkHeight = actualBoundingBoxAscent + actualBoundingBoxDescent;
+      // The x position of the word
+      let startX = textX;
+
+      // Consider edge cases
+      const top = Math.max(gridY, gridY + textY - linkHeight / 2);
+      const bottom = Math.min(
+        top + linkHeight,
+        gridHeight - horizontalBarHeight
+      );
+
+      // Loop through the blocks of text
+      for (let i = 0; i < tokenizedText.length; i += 1) {
+        const { v: value } = tokenizedText[i];
+        let { width: linkWidth } = context.measureText(value);
+
+        // If the text is not visible (right side is less than gridX)
+        if (startX + linkWidth < gridX) {
+          startX += linkWidth;
+
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+
+        // Check if the x position is less than the grid x, then startX will be negative so the linkWidth should be shifted
+        linkWidth += startX < gridX ? startX : 0;
+
+        const left = Math.max(gridX, gridX + startX);
+        const right = Math.min(left + linkWidth, gridWidth - verticalBarWidth);
+
+        // If the block is not a url, continue
+        if (tokenizedText[i].t !== 'url') {
+          startX = right;
+
+          // eslint-disable-next-line no-continue
+          continue;
+        }
```

Comment:
Should I just change the `tokensForText` function to be `findLinksInText` or something like that instead and cache the `linkify.find` results instead? I don't think `tokensForText` would be used anywhere else then.

Reply:
That sounds reasonable to me. [This comment](https://github.com/deephaven/web-client-ui/pull/1088#discussion_r1110448977) is what I was suggesting I believe.

So change to `linksInText` or `findLinksInText` and cache that instead of `tokensForText`